File: src/infrastructure/entities/Brand.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, Unique } from "typeorm";

@Entity("brands")
@Unique(["slug"])
export default class BrandEntity {
  @PrimaryGeneratedColumn("increment", { type: "bigint" })
  id: number;

  @Column({ type: "varchar", length: 100, nullable: false })
  name: string;

  @Column({ type: "varchar", length: 100, unique: true, nullable: false })
  slug: string;

  @Column({ type: "text", nullable: true })
  description?: string;

  @Column({ type: "varchar", length: 255, nullable: true })
  logoUrl?: string;

  @Column({ type: "varchar", length: 255, nullable: true })
  website?: string;
}

File: src/main/java/com/example/demo/brand/entity/BrandEntity.java
package com.example.demo.brand.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@Table(name = "brands")
@NoArgsConstructor
public class BrandEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    @Column(name = "name", columnDefinition = "VARCHAR(100)", nullable = false)
    String name;
    @Column(name = "slug", columnDefinition = "VARCHAR(100)", nullable = false, unique = true)
    String slug;
    @Column(name = "description", columnDefinition = "TEXT")
    String description;
    @Column(name = "logo_url", columnDefinition = "VARCHAR(255)")
    String logoUrl;
    @Column(name = "website", columnDefinition = "VARCHAR(255)")
    String website;
}
